### ISSUE TITLE
wrong 'last-modified'

### ISSUE THREAD

**Issue by rexshihaoren** [PyGithub/PyGithub#629]
the 'last-modified' of a file should be when the file was last updated not when the whole repo was updated

**Comment by peterkline** [comment]
Can you provide a code of what you are running.  How are you accessing a file object?

**Comment by rgerkin** [comment]
The fix in #1032 works and all tests should pass if the build is restarted.  The only failing tests (after the second commit) were not relevant to this PR and have since been fixed in master..

**Comment by Dreadnut93** [comment]
This issue is still present in the version 1.52.

**Comment by rgerkin** [comment]
Yes, I fixed it last year in #1032 but it has not yet been merged.

**Comment by Dreadnut93** [comment]
Can we then reopen the issue until it is merged?

**Comment by 2start** [comment]
still not working...

**Comment by peterkline-es** [comment]
It looks like the fix was never merged as the PR didn't pass tests .  Closed as stale not fixed.

**Comment by rgerkin** [comment]
It was always working.  The problem was always in the unit tests.  Many users are now just using my commit as a hotfix since they want dates to work.

**Comment by yigangzhang** [comment]
Can we please reopen this issue? Or open a new issue to fix it? Still NOT working in 1.55. 
We use this library on prod to retrieve files and sort them by last modified time. Took us hours to troubleshoot that last modified time is the same for all files because it's the repo time.
Now we have to append modify time to filename or file content. 
It's a good library but it's bit painful because of this bug.

**Comment by AntonOfTheWoods** [comment]
@rgerkin , can you expand on what you mean by "The problem was always in the unit tests.". You mean that the repo unit tests are broken, or that you don't care that the tests for your fix don't work?

**Comment by rgerkin** [comment]
I mean that the unit tests are broken.  I would definitely care if my fix was responsible for making good tests fail.

**Comment by Markel564** [comment]
2024 and still not working ...

**Comment by EnricoMi** [comment]
That `last-modified` attribute is a property of all github objects and is coming from the HTTP response. You are here referring to the last-modified timestamp of a git file. I think, that should be a different property like `file_last_modified`, to avoid conflicting expectations. Happy to review a PR.

**Comment by EnricoMi** [comment]
Looks like the proposed fix
```
        commits = self.repository.get_commits(path=self.path)
        if commits.totalCount:
            self._last_modified = self._makeStringAttribute(
                str(commits[0].commit.author.date)
            )
        else:
            self._last_modified = github.GithubObject.NotSet
        return self._last_modified.value
```
is more of a convenience method rather than something Github supports.

**Comment by donatelloOo** [comment]
It looks a good workaround for users is to call `commit.complete()` before retrieving the `commit.last_modified` attribute.
